test/FileWriterTest.php

```diff
@@ -66,4 +68,49 @@ private function getTargetFile()
     {
         return sys_get_temp_dir() . '/' . uniqid('test_', true) . '.php';
     }
+
+    public function testMultipleWriters()
+    {
+        $processes = [];
+        $descriptorSpec = [
+            0 => ['pipe', 'r'],
+            1 => ['pipe', 'w'],
+            2 => ['pipe', 'w'],
+        ];
+
+        for ($i = 0; $i < 20; ++$i) {
+            $processes[$i]['pipes'] = [];
+            $processes[$i]['process'] = proc_open(
+                'php ' . __DIR__ . '/writer.php',
+                $descriptorSpec,
+                $processes[$i]['pipes']
+            );
+        }
+
+        for ($i = 20; $i < 80; ++$i) {
```

Comment:
Should multiple reader processes be started before writes too?

Reply:
I've changed approach, to run one reader process - reading the same file in the loop for 3 seconds and 20 writer processes which start randomly after 0.5-2s.

Added test also with "standard-writer", and using the same reader - and this test should fail.